src/cli/init.js

```diff
@@ -0,0 +1,158 @@
+// @flow
+// Implementation of `sourcecred init`
+
+import dedent from "../util/dedent";
+import {type RepoId, stringToRepoId} from "../core/repoId";
+import {type Project, projectToJSON} from "../core/project";
+import type {Command} from "./command";
+import * as Common from "./common";
+import fs from "fs-extra";
+import process from "process";
+import read from "read";
+import path from "path";
+import {fetchGithubOrg} from "../plugins/github/fetchGithubOrg";
+import {type DiscourseServer} from "../plugins/discourse/loadDiscourse";
+
+function usage(print: (string) => void): void {
+  print(
+    dedent`\
+    usage: sourcecred init
+           sourcecred init --help
+
+    Sets up a new SourceCred instance, by interactively creating a SourceCred
+    project configuration, and saving it to 'sourcecred.json'.
+
+    Arguments:
+        --help
+            Show this help message and exit, as 'sourcecred help init'.
+
+    Environment variables:
+        SOURCECRED_GITHUB_TOKEN
+            API token for GitHub. This should be a 40-character hex
+            string. Required if you want to load whole GitHub orgs.
+
+            To generate a token, create a "Personal access token" at
+            <https://github.com/settings/tokens>. When loading data for
+            public repositories, no special permissions are required.
+            For private repositories, the 'repo' scope is required.
+    `.trimRight()
+  );
+}
+
+function die(std, message) {
+  std.err("fatal: " + message);
+  std.err("fatal: run 'sourcecred help init' for help");
+  return 1;
+}
+
+const initCommand: Command = async (args, std) => {
```

Comment:
I see 3 concerns complected this function.
- cli command handling
- a `(...) => Promise<Project>` dialog
- I/O

Suggest separating these. Which would allow testing selectively, such as just the I/O when the dialog is mocked.